packages/engine/Source/Core/FpsTracker.js

```diff
@@ -0,0 +1,40 @@
+import destroyObject from "../Core/destroyObject.js";
+import getTimestamp from "../Core/getTimestamp.js";
+
+/**
+ * @private
+ */
+function FpsTracker(options) {
+  this._timeStamps = [];
+}
+
+/**
+ * Calculates average fps based on timeStamps of recently rendered frames
+ */
+FpsTracker.prototype.getAverageFps = function () {
+  if (this._timeStamps.length !== 10) {
+    return;
+  }
+  const elapsedTime = (this._timeStamps[0] - this._timeStamps[9]) / 9;
+  const fps = 1 / (elapsedTime / 1000);
+  return fps;
+};
+
+/**
+ * Adds the timestamp for the current frame to the queue
+ */
+FpsTracker.prototype.updateFrameTimeStamp = function () {
+  const time = getTimestamp();
+  this._timeStamps.unshift(time);
+  if (this._timeStamps.length > 10) {
+    this._timeStamps.pop();
+  }
+};
+
+/**
+ * Destroys the WebGL resources held by this object.
```

Comment:
This does not have WebGL resources

Reply:
thank you, since we will now use the FrameRateMonitor class this is no longer relevant